From 19311bfa7b8e8c6effa9c375de9b0eb4338bee12 Mon Sep 17 00:00:00 2001
From: Roman Borisov <ext-roman.borisov@nokia.com>
Date: Thu, 24 Mar 2011 12:08:43 +0000
Subject: mount: add --rbind option support

Signed-off-by: Roman Borisov <ext-roman.borisov@nokia.com>
Signed-off-by: Denys Vlasenko <vda.linux@googlemail.com>
---
(limited to 'util-linux/mount.c')

diff --git a/util-linux/mount.c b/util-linux/mount.c
index e672a73..722d0be 100644
--- a/util-linux/mount.c
+++ b/util-linux/mount.c
@@ -181,6 +181,7 @@ static const int32_t mount_options[] = {
 		/* "relatime"    */ MS_RELATIME,
 		/* "norelatime"  */ ~MS_RELATIME,
 		/* "loud"        */ ~MS_SILENT,
+		/* "rbind"       */ MS_BIND|MS_RECURSIVE,
 
 		// action flags
 		/* "union"       */ MS_UNION,
@@ -236,6 +237,7 @@ static const char mount_option_str[] =
 		"relatime\0"
 		"norelatime\0"
 		"loud\0"
+		"rbind\0"
 
 		// action flags
 		"union\0"
--
cgit v0.9.0.1-2-gef13
